### DISCUSSION TITLE
Logging not working

### DISCUSSION THREAD

**Discussion by sorter** [django/channels#1648] [Q&A]
Hello.

I have an issue where logging statements are not working in subclasses of `WebsocketConsumer`. Logger works in the entry point to the daphne application and even above the class declaration but not within the consumer. 

```
logger = logging.getLogger(__name__)
logger.info("This log shows up")

class TrackNotifyConsumer(WebsocketConsumer):
    """API Tracking Notifications Consumer"""

    def connect(self):
        logger.info("This log DOES NOT show up")
        async_to_sync(self.channel_layer.group_add)("track", self.channel_name)
        self.accept()
```

- Your OS and runtime environment, and browser if applicable
Ubuntu 18.04.4 LTS

- A `pip freeze` output showing your package versions
[freeze.txt](https://github.com/django/channels/files/5927896/freeze.txt)

- What you expected to happen vs. what actually happened
Expected logging to work normally and for logs to show up wherever I invoke `logger.info(...)`

- How you're running Channels (runserver? daphne/runworker? Nginx/Apache in front?)
Daphne, nginx

**Comment by jesseinit** [comment]
I will really appreciate a solution to this.

How are you able to get the logger in settings.py to work. Mine doesn't work. @sorter

**Comment by aryaniyaps** [comment]
Hello @sorter 
Sorry for the late reply.
the logger's default level is `NOTSET`, which means it checks with its parent, which is the root, which has a default of `WARNING`.
And you can probably leave the handler at its default of `NOTSET`, which means it defers to the logger's filtering.

As the connect function is called within a try and except block [already](https://github.com/django/channels/blob/main/channels/generic/websocket.py#L39), I guess warnings are being discarded. Can you try this?
```py
logger.setLevel(logging.INFO)
```
I will try and fix it if it doesn't as soon as I get time. Don't think it is an issue with channels.
https://stackoverflow.com/a/21105825
